src/process/execution_outcomes.rs

```diff
@@ -0,0 +1,77 @@
+use diesel::r2d2::{ConnectionManager, Pool};
+use diesel::PgConnection;
+use tokio_diesel::AsyncRunQueryDsl;
+use tracing::error;
+
+use crate::models;
+use crate::schema;
+
+/// Saves ExecutionOutcome to database and then saves ExecutionOutcomesReceipts
+pub(crate) async fn process_execution_outcomes(
+    pool: &Pool<ConnectionManager<PgConnection>>,
+    execution_outcomes: Vec<&near_indexer::near_primitives::views::ExecutionOutcomeWithIdView>,
+) {
+    let mut outcome_models: Vec<models::execution_outcomes::ExecutionOutcome> = vec![];
+    let mut outcome_receipt_models: Vec<models::execution_outcomes::ExecutionOutcomeReceipt> =
+        vec![];
+    for outcome in execution_outcomes {
+        let model = models::execution_outcomes::ExecutionOutcome::from(outcome);
+        outcome_models.push(model);
+
+        outcome_receipt_models.extend(outcome.outcome.receipt_ids.iter().enumerate().map(
+            |(index, receipt_id)| models::execution_outcomes::ExecutionOutcomeReceipt {
+                execution_outcome_receipt_id: outcome.id.to_string(),
+                index: index as i32,
+                receipt_id: receipt_id.to_string(),
+            },
+        ));
+    }
+
+    loop {
+        match diesel::insert_into(schema::execution_outcomes::table)
+            .values(outcome_models.clone())
+            .on_conflict_do_nothing()
+            .execute_async(&pool)
+            .await
+        {
+            Ok(_) => break,
+            Err(async_error) => {
+                if crate::process::break_on_foreignkey_violation(&async_error) {
+                    break;
+                }
```

Comment:
Do we still need it even with `on_conflict_do_nothing`? It seems suboptimal as a single "broken" execution outcome will lead to skipping all the execution outcomes. Thus, you need to try `on_conflict(on_constraint(...)).do_nothing()`.